tools/build/src/devfile/meta-yaml-to-devfile-yaml.ts

```diff
@@ -179,6 +179,14 @@ export class MetaYamlToDevfileYaml {
       devfileYaml.events = events;
       components = components.concat(componentsFromContainer);
     }
+
+    // remove duplicated components, e.g. plugins
+    // container and init container may provide two volumes with the same name
+    // and it is enough to have in components list only one volume definition
+    components = components.filter(function (item, pos, self) {
```

Comment:
devfile.components = [...new Set(components)]

Reply:
Have tried this solution, didn't work, it didn't remove the duplicate from the list, I suppose because of inlined object `{}`.